src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -491,22 +552,40 @@ internal void CloseExtensionControl(IViewExtension viewExtension)
         /// <param name="e"></param>
         internal void CloseExtensionTab(object sender, RoutedEventArgs e)
         {
-            string tabName = (sender as Button).DataContext.ToString();
-            TabItem tabitem = ExtensionTabItems.OfType<TabItem>().SingleOrDefault(n => n.Header.ToString() == tabName);
+            string tabId = (sender as Button).Uid.ToString();
+            TabItem tabitem = ExtensionTabItems.OfType<TabItem>().SingleOrDefault(n => n.Uid.ToString() == tabId);
 
             if (tabitem.Tag is ViewExtensionBase viewExtensionBase)
             {
                 viewExtensionBase.Closed();
             }
 
+            dynamoViewModel.DockedWindows.TryGetValue(tabitem.Uid, out NodeModel nodeModel);
```

Comment:
Instead of grabbing the node model, it seems we can simply search the node using the uid?

Reply:
yeah good point, search the nodes in the workspace with the ID?

Reply:
Using just a HashSet now to get the current docked windows. Not storing the nodemodel object now and getting that info from workspace nodes.